### PR TITLE
Fix rustc/clippy warnings

### PR DESCRIPTION
**Pull request by regexident**
Before: 400 warnings
After: 6 warnings

### CODE REVIEW

File: sctp/src/association/association_test.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// Silence warning on `for i in 0..vec.len() { … }`:
+#![allow(clippy::needless_range_loop)]
+
```
Comment (algesten) [inline sctp/src/association/association_test.rs:3]: Here

File: sctp/src/timer/timer_test.rs [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// Silence warning on `for i in 0..vec.len() { … }`:
+#![allow(clippy::needless_range_loop)]
+
```
Comment (algesten) [inline sctp/src/timer/timer_test.rs:3]: yo

File: webrtc/src/data_channel/data_channel_test.rs [hunk 3]
```diff
@@ -1,3 +1,6 @@
+// Silence warning on `for i in 0..vec.len() { … }`:
+#![allow(clippy::needless_range_loop)]
+
```
Comment (algesten) [inline webrtc/src/data_channel/data_channel_test.rs:3]: here

File: webrtc/src/data_channel/data_channel_test.rs [hunk 4]
```diff
@@ -882,8 +872,8 @@ async fn test_data_channel_buffered_amount_set_before_open() -> Result<()> {
         let dc3 = Arc::clone(&dc2);
         Box::pin(async move {
             for _ in 0..10 {
-                if let Err(_) = dc3.send(&buf).await {
-                    assert!(false, "Failed to send string on data channel");
+                if dc3.send(&buf).await.is_err() {
+                    panic!("Failed to send string on data channel");
```
Comment (algesten) [inline webrtc/src/data_channel/data_channel_test.rs:876]: This could be `.expect()`

File: webrtc/src/data_channel/data_channel_test.rs [hunk 5]
```diff
@@ -982,8 +972,8 @@ async fn test_data_channel_buffered_amount_set_after_open() -> Result<()> {
             .await;
 
             for _ in 0..10 {
-                if let Err(_) = dc3.send(&buf).await {
-                    assert!(false, "Failed to send string on data channel");
+                if dc3.send(&buf).await.is_err() {
+                    panic!("Failed to send string on data channel");
```
Comment (algesten) [inline webrtc/src/data_channel/data_channel_test.rs:976]: This could be `.expect()`

**Review by algesten — CHANGES_REQUESTED** [review]
Major work! Thanks!

Some little niggly things I picked up on.

File: dtls/src/conn/conn_test.rs [hunk 6]
```diff
@@ -1983,15 +1962,7 @@ async fn test_server_timeout() -> Result<()> {
...
     .await;
-    if let Err(err) = result {
-        assert!(
-            true,
-            "Sever error exp(Temporary network error) failed({})",
-            err
-        );
-    } else {
-        assert!(false, "Expected Error but got Ok");
-    }
```
[the remark this answers is not in the document]
Comment (regexident) [reply dtls/src/conn/conn_test.rs:1994]: No clue, tbh. My guess would be on documentation purposes?

File: dtls/src/conn/conn_test.rs [hunk 7]
```diff
@@ -2507,23 +2476,18 @@ async fn test_renegotation_info() -> Result<()> {
...
 
         let mut got_negotation_info = false;
         for v in &server_hello.extensions {
-            match v {
-                Extension::RenegotiationInfo(_) => {
-                    got_negotation_info = true;
-                }
-                _ => {}
+            if let Extension::RenegotiationInfo(_) = v {
+                got_negotation_info = true;
```
[the remark this answers is not in the document]
Comment (regexident) [reply dtls/src/conn/conn_test.rs:2490]: Even, even nicer here is

```rust
let got_negotation_info = server_hello
    .extensions
    .iter()
    .any(|v| matches!(v, Extension::RenegotiationInfo(_)));
```

😄

File: dtls/src/handshake/handshake_message_client_hello/handshake_message_client_hello_test.rs [hunk 8]
```diff
@@ -41,8 +41,8 @@ fn test_handshake_message_client_hello() -> Result<()> {
         cipher_suites: vec![
             CipherSuiteId::Tls_Ecdhe_Ecdsa_With_Aes_128_Gcm_Sha256,
             CipherSuiteId::Tls_Ecdhe_Ecdsa_With_Aes_256_Cbc_Sha,
-            //Box::new(CipherSuiteTLSEcdheEcdsaWithAes128GcmSha256::default()),
-            //Box::new(CipherSuiteTLSEcdheEcdsaWithAes256CbcSha::default()),
+            //Box::<CipherSuiteTLSEcdheEcdsaWithAes128GcmSha256>::default(),
+            //Box::<CipherSuiteTLSEcdheEcdsaWithAes256CbcSha>::default(),
```
[the remark this answers is not in the document]
Comment (regexident) [reply dtls/src/handshake/handshake_message_client_hello/handshake_message_client_hello_test.rs:45]: It resolves this clippy warning:
https://rust-lang.github.io/rust-clippy/master/#box_default

File: interceptor/src/stats/interceptor.rs [hunk 9]
```diff
@@ -821,17 +821,16 @@ fn calculate_rtt_ms(now: u32, delay: u32, last_report: u32) -> Option<f64> {
 
 #[cfg(test)]
 mod test {
+    // Silence warning on `..Default::default()` with no effect:
+    #![allow(clippy::needless_update)]
+
```
[the remark this answers is not in the document]
Comment (regexident) [reply interceptor/src/stats/interceptor.rs:826]: I added that to make it more obvious that the attribute applies to the entire scope (i.e. `#![…]`), not just the macro that follows (i.e. `#[…]`). The `!` is easy to miss while glancing over.

[the remark this answers is not in the document]
Comment (regexident) [reply interceptor/src/stats/interceptor.rs:826]: Also the existing module-scope `#![…]` directive blocks (~30) all are separated by a whitespace line. 🤔

File: mdns/src/message/message_test.rs [hunk 10]
```diff
@@ -1,3 +1,6 @@
+// Silence warning on complex types:
+#![allow(clippy::type_complexity)]
+
```
[the remark this answers is not in the document]
Comment (regexident) [reply mdns/src/message/message_test.rs:3]: https://github.com/webrtc-rs/webrtc/pull/384#discussion_r1061222916

File: rtp/benches/packet_bench.rs [hunk 11]
```diff
@@ -1,3 +1,6 @@
+// Silence warning on `..Default::default()` with no effect:
+#![allow(clippy::needless_update)]
+
```
[the remark this answers is not in the document]
Comment (regexident) [reply rtp/benches/packet_bench.rs:3]: https://github.com/webrtc-rs/webrtc/pull/384#discussion_r1061222916

File: rtp/src/codecs/h264/h264_test.rs [hunk 12]
```diff
@@ -1,3 +1,6 @@
+// Silence warning on `for i in 0..vec.len() { … }`:
+#![allow(clippy::needless_range_loop)]
+
```
[the remark this answers is not in the document]
Comment (regexident) [reply rtp/src/codecs/h264/h264_test.rs:3]: https://github.com/webrtc-rs/webrtc/pull/384#discussion_r1061222916

File: webrtc/src/data_channel/data_channel_test.rs (hunk 3 above)
Comment (regexident) [reply webrtc/src/data_channel/data_channel_test.rs:3]: https://github.com/webrtc-rs/webrtc/pull/384#discussion_r1061222916

File: interceptor/src/stats/interceptor.rs (hunk 9 above)
[the remark this answers is not in the document]
Comment (algesten) [reply interceptor/src/stats/interceptor.rs:826]: Ok

**Comment by algesten** [comment]
I've resolved the comments that I don't think need addressing.

File: dtls/src/handshake/handshake_message_client_hello/handshake_message_client_hello_test.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (k0nserv) [reply dtls/src/handshake/handshake_message_client_hello/handshake_message_client_hello_test.rs:45]: Weird that it applies to commented out code 🤔

**Review by k0nserv — COMMENTED** [review]
Great PR @regexident!

If we get to zero we could make CI enforce that

**Comment by regexident** [comment]
> Great PR @regexident!
> 
> If we get to zero we could make CI enforce that

@k0nserv 💯 !

The remaining warnings are instances of `clippy::identity_op`.

1. `(1 << 16) | 0x0000`:

    https://github.com/webrtc-rs/webrtc/blob/958c0c44dc77f1f29ac3933b5b551b82aadff1e8/interceptor/src/report/receiver/receiver_test.rs#L429-L440

    Not even sure what this was supposed to do?

2. `(256u16 * 1 / 3) as u8`

    https://github.com/webrtc-rs/webrtc/blob/958c0c44dc77f1f29ac3933b5b551b82aadff1e8/interceptor/src/report/receiver/receiver_test.rs#L429-L440

    https://github.com/webrtc-rs/webrtc/blob/958c0c44dc77f1f29ac3933b5b551b82aadff1e8/interceptor/src/report/receiver/receiver_test.rs#L540-L551

    These ones should probably be changed to `(256u16 / 3) as u8`, I suppose?

3. `id * 1`

    https://github.com/webrtc-rs/webrtc/blob/958c0c44dc77f1f29ac3933b5b551b82aadff1e8/interceptor/src/twcc/sender/sender_test.rs#L39

    This one seems to be intentional for syntactical consistency, so putting a `#[allow(clippy::identity_op)]` in front of it seems like a reasonable approach.

**Comment by k0nserv** [comment]
Unsure about **1**, your link for it is wrong.

**2** can just be `256 / 3` yes, alternatively as explained [here](https://www.rfc-editor.org/rfc/rfc3550#appendix-A.3) it's `(1 << 8) / 3` which yields the same thing but is more akin to what the spec says.

Agreed on **3**

**Comment by regexident** [comment]
> Unsure about **1**, your link for it is wrong.

Fixed!

https://github.com/webrtc-rs/webrtc/blob/958c0c44dc77f1f29ac3933b5b551b82aadff1e8/interceptor/src/report/receiver/receiver_test.rs#L257-L268

> **2** can just be `256 / 3` yes, alternatively as explained [here](https://www.rfc-editor.org/rfc/rfc3550#appendix-A.3) it's `(1 << 8) / 3` which yields the same thing but is more akin to what the spec says.

Will make it `(1 << 8) / 3` then.

> Agreed on **3**

:+1:

**Comment by k0nserv** [comment]
Ah, the high 16 bits are the number of cycles and the low 16 bits is the current sequence number, that's why **1** is written that way. I think we ignore the clippy error there too for the same reason as **3**.

**Comment by regexident** [comment]
The warning is about the `… | 0x0000`, which has no effect.

And unlike #3 there doesn't seem to be any such pattern for the values provided to `last_sequence_number: …`, afaict.

**Comment by k0nserv** [comment]
I think the `0x0000` bit is making it clear that we are setting both the high and low bits. We could send an extra packet with sequence number 1 in the test so it becomes

```rust
(1 << 16) | 0x0001
```

Clippy is right that it has no effect but it help with clarity to keep it this way I think, especially if we add a comment

**Review by k0nserv — APPROVED** [review]

**Review by melekes — APPROVED** [review]

**Comment by regexident** [comment]
@algesten looks like it's stuck on your change request, which should all have been resolved. 🤔

**Review by algesten — APPROVED** [review]
Ship it! Great work!

**Pull request merged**